$NetBSD$

* Add targets for object file, shared object and test.
* Honor CPPFLAGS, CFLAGS and LDFLAGS

--- Makefile.orig	2021-07-14 10:28:45.000000000 +0000
+++ Makefile
@@ -1,8 +1,16 @@
-all: sds-test
+all: sds-test libsds.so
+
+sds.o: sds.c sds.h
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c -fpic -Wall -std=c99 -pedantic -O2 sds.c
+
+libsds.so: sds.o
+	$(CC) $(LDFLAGS) -shared -o libsds.so sds.o
 
 sds-test: sds.c sds.h testhelp.h
-	$(CC) -o sds-test sds.c -Wall -std=c99 -pedantic -O2 -DSDS_TEST_MAIN
-	@echo ">>> Type ./sds-test to run the sds.c unit tests."
+	$(CC) $(CPPFLAGS) $(CFLAGS) -o sds-test sds.c -Wall -std=c99 -pedantic -O2 -DSDS_TEST_MAIN
+
+test:
+	./sds-test
 
 clean:
-	rm -f sds-test
+	rm -f sds-test sds.o libsds.so
